### DISCUSSION TITLE
Chapter_2

### DISCUSSION THREAD

**Discussion by hongsam14** [Kookmin-Univ-C-programming/SnakeGame#17] [General]
2주차: 챕터_2

**Comment by hongsam14** [comment]
![1](https://user-images.githubusercontent.com/69339846/170041417-b9d29dfa-77da-4f15-af3d-acacff7f172a.JPG)

**Comment by hongsam14** [comment]
![2](https://user-images.githubusercontent.com/69339846/170041453-7e3b4061-031b-482d-8389-a1134eb89400.JPG)

**Comment by hongsam14** [comment]
![3](https://user-images.githubusercontent.com/69339846/170041574-6866d289-36ac-419d-a6bc-d04c7c942a57.JPG)

**Comment by hongsam14** [comment]
![4](https://user-images.githubusercontent.com/69339846/170041687-21b903f2-1396-48fd-a87d-db56af18aecc.JPG)

**Comment by hongsam14** [comment]
![5](https://user-images.githubusercontent.com/69339846/170041721-b2a0b888-2012-4443-8de0-eb0d519b4197.JPG)

**Comment by hongsam14** [comment]
![6](https://user-images.githubusercontent.com/69339846/170041752-65607c6a-4df7-4fc2-8722-2337c2b41744.JPG)

**Comment by hongsam14** [comment]
임무 분담을 해야겠군요.

**Reply by mumat0103** [reply]
그렇죠 나눠야겠네요

**Reply by hongsam14** [reply]
좀 생각을 해봐야겠네요. 같은 파일 만지게 되면 comflict나겠죠 ㅋㅋ. 확실하게 나누면 그럴 일이 없겠지만요.

**Reply by mumat0103** [reply]
아무래도 같은 파일 만지면 충돌이 일어나겠죠 ㅋㅋㅋ 어떻게 나눠야할까요

**Reply by hongsam14** [reply]
당장 생각나는건 Snake객체 구현과 Game 메인함수 구현 이렇게 나눠질 거 같긴 한데...

**Reply by hongsam14** [reply]
게임 부분은 아마 초당 프레임 구현하는거랑 키입력 받는 것 이렇게 일거 같고...

**Reply by mumat0103** [reply]
그렇죠...? 게임부분은 틱 구현하는 거랑 키 입력 받는 거 일테고.. Snake 객체는 늘어나는 것까지 생각해서 짜는 게 좋을까요?

**Reply by hongsam14** [reply]
스네이크 객체 부분은 게임필드 부분이랑 같이 만질듯요. 만들때 GameRule#2도 어느 정도 고려 해야할 것 같고요

**Reply by hongsam14** [reply]
네네 늘고 줄고도 고려해야할 것 같습니다.

**Reply by hongsam14** [reply]
둘 다 다른 의미로 난이도 있을 듯 ㅋㅋ

**Reply by hongsam14** [reply]
게임 메인 부분은 여기서 같이 고려해봐야할듯요.

**Reply by hongsam14** [reply]
코드 짜는 것과는 별개로 이부분 고민하다보면 스네이크에서 필요한 멤버함수도 나올 것 같아요.

**Reply by mumat0103** [reply]
하고 싶은 부분 있으신가요??ㅋㅋ

**Reply by mumat0103** [reply]
그쵸 고민하다보면 이것저것 많이 나올 것 같네요

**Comment by mumat0103** [comment]
1. 키 입력
2. 스네이크 객체의 이동
3. 맵에서의 값 변경
4. 스크린을 하나 만들어서, 게임필드 값을 계속 받아온다.
5. 344를 위치시키는 것이 이동.
6. 방향 확인 후 체인지. 멤버함수 호출. 마지막 설정한 방향으로 쭉쭉
7. 틱 당 키 입력을 받아서, 빈 공간이면 진행, 빈 공간이 아니면 실패.
8. game init, snake init, game data 상정. 이벤트를 받아서 상하좌우 키 구현, 틱 당 루프 구현,

게임 루프 담당자 = 스네이크 콜리전 데이터, 기존의 필드 비교 함수, 키 입력 받는 함수.
스네이크 = 자기의 몸통과 머리 데이터(좌표), 스택 같은 자료구조, move를 통해서 원하는 좌표로 이동, 부딪혔을 때, 먹을 때 데이터 변경.

**Reply by hongsam14** [reply]
필드 비교함수는 게임 필드 객체에서 메소드로 구현해도 될듯요.
